src/Nethermind/Nethermind.Blockchain/Processing/BlockProcessor.cs

```diff
@@ -292,10 +239,38 @@ private void ValidateProcessedBlock(Block suggestedBlock, ProcessingOptions opti
         }
 
         // TODO: block processor pipeline
-        protected virtual TxReceipt[] ProcessBlock(Block block, IBlockTracer blockTracer, ProcessingOptions options)
+        protected virtual TxReceipt[] ProcessBlock(
+            Block block, 
+            IBlockTracer blockTracer, 
+            ProcessingOptions options,
+            ITransactionProcessingStrategy? transactionProcessingStrategy = null)
         {
             IReleaseSpec spec = _specProvider.GetSpec(block.Number);
-            TxReceipt[] receipts = ProcessTransactions(block, options, blockTracer, spec);
+            
+            _receiptsTracer.SetOtherTracer(blockTracer);
+            _receiptsTracer.StartNewBlockTrace(block);
+
+            if (transactionProcessingStrategy == null) 
+            {
+                if (block is BlockToProduce {Transactions: not ICollection<Transaction>})
+                {
+                    transactionProcessingStrategy = new ProducingBlockTransactionProcessingStrategy(
```

Comment:
strategy should be injected in constructor, then we can drop BlockToProduce alltogether

Reply:
This is in scope of next PR